robot/impl/local_robot.go

```diff
@@ -52,6 +52,7 @@ type localRobot struct {
 	cloudConnSvc               cloud.ConnectionService
 	logger                     golog.Logger
 	activeBackgroundWorkers    sync.WaitGroup
+	reconfigureWorkers         sync.WaitGroup
```

Comment:
Should we also wait on this wait group in `Close`? @dgottlieb I wonder if that will introduce data races with https://github.com/viamrobotics/rdk/pull/2789 though.

Reply:
I don't think we need to do anything formal with the `reconfigureWorkers` in `Close`. If in the future more things need to care about the lifetime of reconfigure workers we can take a second look if embedding a `Wait` somewhere makes sense.

Reply:
Thanks, SGTM!